src/ui/public/visualize/visualize.js

```diff
@@ -147,49 +148,22 @@ uiModules
         $scope.vis.on('reload', reload);
         // auto reload will trigger this event
         $scope.$on('courier:searchRefresh', reload);
-        // dashboard will fire fetch event when it wants to refresh
-        $scope.$on('fetch', reload);
-
 
-
-        const handleQueryUpdate = ()=> {
-          $scope.fetch();
-        };
-        queryFilter.on('update', handleQueryUpdate);
-
-        if ($scope.appState) {
-          const stateMonitor = stateMonitorFactory.create($scope.appState);
-          stateMonitor.onChange((status, type, keys) => {
-            if (keys[0] === 'vis') {
-              if ($scope.appState.vis) $scope.vis.setState($scope.appState.vis);
-              $scope.fetch();
-            }
-            if ($scope.vis.type.requiresSearch && ['query', 'filters'].includes(keys[0])) {
-              $scope.fetch();
-            }
-          });
-
-          $scope.$on('$destroy', () => {
-            stateMonitor.destroy();
-          });
-        }
+        $scope.$watchGroup(['filters', 'query', 'timeRange'], fetch);
```

Comment:
Shouldn't that be `$scope.fetch`?

Reply:
oh .... i guess thats the reason watch is not triggering :D lol, thanks!